pkg/store/postings_codec.go

```diff
@@ -58,6 +57,67 @@ func isDiffVarintSnappyStreamedEncodedPostings(input []byte) bool {
 	return bytes.HasPrefix(input, []byte(codecHeaderStreamedSnappy))
 }
 
+var (
+	readersPool = sync.Pool{
+		New: func() interface{} {
+			return s2.NewReader(nil, s2.ReaderAllocBlock(1<<10))
```

Comment:
If you only have snappy compatible blocks they cannot be more than 64KB:

```suggestion
			return s2.NewReader(nil, s2.ReaderAllocBlock(1<<10), s2.ReaderMaxBlockSize(64<<10))
```